### PR TITLE
fix: makes metrics append-only & resolves deadlock on listener timeouts

### PR DESCRIPTION
**Pull request by corban-beaird**
## Description
Makes metrics append-only, and resolves deadlock issue on pq.Listener timeouts 

## Test Plan
- Run existing integration tests 

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by corban-beaird** [comment]
Before this gets merged, I'll add additional integration test cases for metrics

File: master/internal/stream/system.go [hunk 1]
```diff
@@ -337,12 +337,12 @@ func (ps *PublisherSet) entrypoint(
 	}()
 
 	streamer := stream.NewStreamer(prepareFunc)
-
 	// read first startup message
 	var startupMsg StartupMsg
 	err := socket.ReadJSON(&startupMsg)
 	if err != nil {
-		return fmt.Errorf("error while reading initial startup message: %s", err.Error())
```
Comment (corban-beaird) [inline master/internal/stream/system.go:345]: I don't believe this change is necessary, will revert

**Comment by eecsliu** [comment]
Looks mostly good, I'll give it another review after tests are added + conflicts resolved

File: master/static/migrations/20231116155837_add-metrics-trigger.tx.up.sql [hunk 2]
```diff
@@ -69,135 +63,28 @@ BEGIN
...
-    trials t ON t.experiment_id = e.id
-        INNER JOIN
-    metrics m ON m.trial_id = t.id
-WHERE
-    p.id = NEW.id
-    LOOP
-        metric.seq = nextval('stream_metric_seq');
-        UPDATE metrics SET seq = metric.seq where id = metric.id;
-        jmetric = to_jsonb(metric);
-        PERFORM stream_metric_notify(jmetric, OLD.workspace_id, OLD.experiment_id, jmetric, NEW.workspace_id, NEW.experiment_id);
```
Comment (eecsliu) [inline master/static/migrations/20231116155837_add-metrics-trigger.tx.up.sql:152]: yaaas delete it 
soothing face small

**Review by eecsliu — APPROVED** [review]
LGTM!

**Pull request merged**